perception/image_projection_based_fusion/src/roi_pointcloud_fusion/node.cpp

```diff
@@ -0,0 +1,199 @@
+// Copyright 2023 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "image_projection_based_fusion/roi_pointcloud_fusion/node.hpp"
+
+#include "image_projection_based_fusion/utils/geometry.hpp"
+#include "image_projection_based_fusion/utils/utils.hpp"
+
+#ifdef ROS_DISTRO_GALACTIC
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <tf2_sensor_msgs/tf2_sensor_msgs.h>
+#else
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+#include <tf2_sensor_msgs/tf2_sensor_msgs.hpp>
+#endif
+
+#include "euclidean_cluster/utils.hpp"
+
+namespace image_projection_based_fusion
+{
+RoiPointCloudFusionNode::RoiPointCloudFusionNode(const rclcpp::NodeOptions & options)
+: FusionNode<PointCloud2, DetectedObjectWithFeature>("roi_pointcloud_fusion", options)
+{
+  fuse_unknown_only_ = static_cast<bool>(declare_parameter("fuse_unknown_only", true));
+  min_cluster_size_ = static_cast<int>(declare_parameter("min_cluster_size", 2));
+  cluster_2d_tolerance_ = declare_parameter<double>("cluster_2d_tolerance", 0.5);
+  pub_objects_ptr_ =
+    this->create_publisher<DetectedObjectsWithFeature>("output_clusters", rclcpp::QoS{1});
+  cluster_debug_pub_ = this->create_publisher<sensor_msgs::msg::PointCloud2>("debug/clusters", 1);
```

Comment:
[imo] To reduce processing load, could you add an 'output_debug' option avoid output during AD driving?